File: examples/tool_permission_plan.ts
export interface ToolRepairOutput {
  summary: string;
  searched: boolean;
  test_passed: boolean;
  edited: boolean;
}

export interface ToolRepairFollowup {
  action: "commit_repair" | "rerun_tests" | "inspect_manually";
  requiresHumanReview: boolean;
  summary: string;
}

export function planRepairFollowup(output: ToolRepairOutput): ToolRepairFollowup {
  if (output.edited && output.test_passed) {
    return {
      action: "commit_repair",
      requiresHumanReview: false,
      summary: output.summary,
    };
  }
  if (output.edited) {
    return {
      action: "rerun_tests",
      requiresHumanReview: true,
      summary: output.summary,
    };
  }
  return {
    action: "inspect_manually",
    requiresHumanReview: true,
    summary: output.summary,
  };
}

File: examples/tool_permission_plan.py
from typing import Any

ToolRepairOutput = dict[str, Any]


def plan_repair_followup(output: ToolRepairOutput) -> dict[str, object]:
    if bool(output["edited"]) and bool(output["test_passed"]):
        return {
            "action": "commit_repair",
            "requires_human_review": False,
            "summary": str(output["summary"]),
        }
    if bool(output["edited"]):
        return {
            "action": "rerun_tests",
            "requires_human_review": True,
            "summary": str(output["summary"]),
        }
    return {
        "action": "inspect_manually",
        "requires_human_review": True,
        "summary": str(output["summary"]),
    }
